ts/src/encoding/customTypes/Dec.ts

```diff
@@ -10,6 +10,6 @@ const PRECISION = 18;
 export const Dec = {
   typeName: "github.com/cosmos/cosmos-sdk/types.Dec",
   shortName: "Dec",
-  encode: (value: string) => Decimal.fromUserInput(value.length ? value : "0", PRECISION).atomics,
-  decode: (value: string) => Decimal.fromAtomics(value.length ? value : "0", PRECISION).toString(),
+  encode: (value: string) => value.length ? Decimal.fromUserInput(value, PRECISION).atomics : "",
+  decode: (value: string) => value.length ? Decimal.fromAtomics(value, PRECISION).toString() : "",
```

Comment:
this is to ensure that Dec.encode and Dec.decode are inverse functions. Previously this rule was broken:

```ts
const value = "";
expect(Dec.decode(Dec.encode(value))).toBe(value);
```